src/Toggle/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import styled from 'styled-components'
+
+export interface ToggleProps {
```

Comment:
something would say might worth exploring for defaults ->
Also why all optional? pretty sure checked could be required?
some docs/ideas:
https://www.typescriptlang.org/docs/handbook/utility-types.html

```ts
type ToggleSize = {
  width: string, // maybe number and then suffix with px?
  height: string,
  padding: string,
  ...
}

interface ToggleProps {
    style: ToggleStyle
    checked: boolean
    ...
}

const defaults: Pick<ToggleProps, 'style'> = {
    width: '300px',
    height: '50px',
    padding: '5px',
    ....
}

const Toggle = (props: ToggleProps = {
    ...defaults,
    checked: false
}) => ...
```

would need to test the above, but something like this I did use in the past

Reply:
This is interesting. I wonder if there's a way of doing this without extra nesting. That's the only thing I'm not sure of. Regarding all optionals I would only make something required if there is no sane default. Otherwise you potentially force someone to pass a prop that they don't care about.

Reply:
What value would be added by doing it this way? The style I've used of optional props and defaults matches the other components in the repo

Reply:
I think for now you're right @jonmattgray. There is a question of sane application of defaults because our current way does look a bit messy. This way is "neat" in a way but I wonder if it's just a bit confusing and abuses the type system to make the code a little neater.